### PR TITLE
Emulating the *file convention

### PR DESCRIPTION
**Pull request by raphaelstolt**
- Suggests naming Phulpfiles similar to Vagrantfile, Boxfile and the like
- Allows easier pattern matching
- Works with non convention following Phulpfiles

### CODE REVIEW

**Comment by reisraff** [comment]
I like the idea of the filename be case insensitive, but I don't like the idea of the file doesn't have the `php` extension, and more... also is not good the pattern to have the asterisk `*` to allow be any character.

And... What if I have in the director:? 

```
/
|-- project/
|   |-- PhulpFile.php
|   |-- phulpfile.php
...
|   |-- phulpFile.php
...
```

**Comment by raphaelstolt** [comment]
As Phulp is a CLI I don't really see the risk in dropping the `php` extension. The pattern is narrowed in the next commit and having __several__ Phulpfiles in one directory seems more like a task/build orchestration. [smell](http://martinfowler.com/bliki/CodeSmell.html) to me.

**Comment by raphaelstolt** [comment]
Btw does a `<commit-msg> (#pr-number)` comment like in your commits auto-close the releated PR? Kind of the missing piece in my `avoid merge commits` worklow.

File: bin/phulp.php [hunk 1]
```diff
@@ -20,11 +20,17 @@
     }
 }
 
-$phulpFile = './PhulpFile.php';
-if ( ! file_exists($phulpFile)) {
+$hasPhulpFile = false;
+$phulpFile = './Phulpfile';
```
Comment (reisraff) [inline bin/phulp.php]: Is not necessary.

File: bin/phulp.php [hunk 2]
```diff
@@ -20,11 +20,17 @@
...
+foreach (glob('[P,p]hulp[Ff]il{e,e.php}', GLOB_BRACE) as $filename) {
+    $phulpFile = $filename;
+    $hasPhulpFile = true;
+}
+
+if ($hasPhulpFile === false) {
     Phulp\Output::out(
         '[' . Phulp\Output::colorize((new \DateTime())->format('H:i:s'), 'light_gray') . ']'
-        .  Phulp\Output::colorize(' The PhulpFile.php was not created.', 'red')
+        .  Phulp\Output::colorize(' The ' . basename($phulpFile) . ' does not exist.', 'red')
```
Comment (reisraff) [inline bin/phulp.php]: Here you can fix for

```php
<?php
// ...
Phulp\Output::err( // out ~> err
    '[' . Phulp\Output::colorize((new \DateTime())->format('H:i:s'), 'light_gray') . '] '
    .  Phulp\Output::colorize('There\'s no the phulpfile.', 'red')
);

Phulp\Output::out( // out
    '[' . Phulp\Output::colorize((new \DateTime())->format('H:i:s'), 'light_gray') . '] '
    .  'Please check the documentation for proper phulpfile formatting.'
);
```

File: bin/phulp.php (hunk 1 above)
Comment (raphaelstolt) [inline bin/phulp.php]: Removing them would produce notices for undefined variables.

File: bin/phulp.php [hunk 3]
```diff
@@ -20,11 +20,17 @@
     }
 }
 
-$phulpFile = './PhulpFile.php';
-if ( ! file_exists($phulpFile)) {
+$hasPhulpFile = false;
+$phulpFile = './Phulpfile';
+foreach (glob('[P,p]hulp[Ff]il{e,e.php}', GLOB_BRACE) as $filename) {
+    $phulpFile = $filename;
+    $hasPhulpFile = true;
```
Comment (reisraff) [inline bin/phulp.php]: Is not necessary

File: bin/phulp.php [hunk 4]
```diff
@@ -20,11 +20,17 @@
...
+}
+
+if ($hasPhulpFile === false) {
```
Comment (reisraff) [inline bin/phulp.php]: change for

```php
<?php

if (isset($phulpfile)) {}
```

File: bin/phulp.php [hunk 5]
```diff
@@ -20,11 +20,17 @@
     }
 }
 
-$phulpFile = './PhulpFile.php';
-if ( ! file_exists($phulpFile)) {
+$hasPhulpFile = false;
```
Comment (reisraff) [inline bin/phulp.php]: Is not necessary.

File: bin/phulp.php (hunk 1 above)
Comment (reisraff) [inline bin/phulp.php]: take a look at all my comments.

**Comment by reisraff** [comment]
And one more thing, if the glob returns more than one result throws a warning with a message for fix it.

**Comment by raphaelstolt** [comment]
Now I want an answer regarding. :smile:

> Btw does a <commit-msg> (#pr-number) comment like in your commits auto-close the releated PR? Kind of the missing piece in my avoid merge commits workflow.

File: bin/phulp.php [hunk 6]
```diff
@@ -20,18 +20,36 @@
...
+    Phulp\Output::err(
+        '[' . Phulp\Output::colorize((new \DateTime())->format('H:i:s'), 'light_gray') . '] '
+        .  Phulp\Output::colorize(' There\'s no Phulpfile present. ', 'red')
+    );
+
     Phulp\Output::out(
-        '[' . Phulp\Output::colorize((new \DateTime())->format('H:i:s'), 'light_gray') . ']'
-        .  Phulp\Output::colorize(' The PhulpFile.php was not created.', 'red')
+        '[' . Phulp\Output::colorize((new \DateTime())->format('H:i:s'), 'light_gray') . '] '
+        .  ' Please check the documentation for proper Phulpfile naming. '
```
Comment (reisraff) [inline bin/phulp.php]: trim this string, cause already have space at the last string. :joy:

File: bin/phulp.php [hunk 7]
```diff
@@ -20,18 +20,36 @@
...
+
+    exit(1);
+}
+
+$phulpFile = array_pop($phulpFiles);
+
+if (!isset($phulpFile)) {
+    Phulp\Output::err(
+        '[' . Phulp\Output::colorize((new \DateTime())->format('H:i:s'), 'light_gray') . '] '
+        .  Phulp\Output::colorize(' There\'s no Phulpfile present. ', 'red')
```
Comment (reisraff) [inline bin/phulp.php]: here also.

File: bin/phulp.php [hunk 8]
```diff
@@ -20,18 +20,36 @@
...
 }
 
-$phulpFile = './PhulpFile.php';
-if ( ! file_exists($phulpFile)) {
+$phulpFiles = glob('[P,p]hulp[Ff]il{e,e.php}', GLOB_BRACE);
+
+if (count($phulpFiles) > 1) {
+    Phulp\Output::err(
+        '[' . Phulp\Output::colorize((new \DateTime())->format('H:i:s'), 'light_gray') . '] '
+        .  Phulp\Output::colorize(' There are more than one Phulpfile present. ', 'red')
```
Comment (reisraff) [inline bin/phulp.php]: here also.

**Comment by reisraff** [comment]
About your feedback can you see in the commit log if was what you wanted.

**Comment by raphaelstolt** [comment]
Which feedback? Can't see if a (#pr-number) in the commit message closes the related PR from the logs. 😉 

Btw noticed that you changed casing in your commit messages.

**Pull request merged**